diff --git a/dlls/compobj.dll16/compobj.c b/dlls/compobj.dll16/compobj.c
index 55be07e2..94dd8000 100644
--- a/dlls/compobj.dll16/compobj.c
+++ b/dlls/compobj.dll16/compobj.c
@@ -106,6 +106,8 @@ static WORD Table_ETask[62];
 
 static SEGPTR compobj_malloc;
 
+HRESULT WINAPI CoCreateStandardMalloc16(MEMCTX context, SEGPTR *malloc);
+
 /* --- IMalloc16 implementation */
 
 
@@ -281,6 +283,8 @@ DWORD WINAPI CoBuildVersion16(void)
  */
 HRESULT WINAPI CoGetMalloc16(MEMCTX context, SEGPTR *malloc)
 {
+    if (!compobj_malloc)
+        CoCreateStandardMalloc16(MEMCTX_TASK, &compobj_malloc);
     call_IMalloc_AddRef(*malloc = compobj_malloc);
     return S_OK;
 }
@@ -307,6 +311,8 @@ SEGPTR WINAPI CoMemAlloc(DWORD size, MEMCTX context, DWORD unknown)
     if (unknown)
         FIXME("Ignoring unknown parameter %#lx.\n", unknown);
 
+    if (!compobj_malloc)
+        CoCreateStandardMalloc16(MEMCTX_TASK, &compobj_malloc);
     return call_IMalloc_Alloc(compobj_malloc, size);
 }
 
